src/Hazelcast.Net/CP/CPDistributedObjectBase.cs

```diff
@@ -66,7 +72,33 @@ protected ValueTask<TObject> ToObjectAsync<TObject>(IData valueData)
                 ? new ValueTask<TObject>(obj)
                 : SerializationService.ToObjectAsync<TObject>(valueData, state);
         }
-        
+
+        /// <summary>
+        /// Sends the invocation to the CP group leader if leader connection exists
+        /// and <see cref="NetworkingOptions.CPDirectToLeaderEnabled"/> enabled and the cluster is Enterprise;
+        /// otherwise sends to any member.
+        /// </summary>
+        /// <returns></returns>
+        protected Task<ClientMessage> SendAsync(ClientMessage request, CPGroupId cpGroupId)
+        {
+            if (Cluster.State.IsEnterprise && Cluster.State.Options.Networking.CPDirectToLeaderEnabled)
+            {
+                var leader = Cluster.Members.GetLeaderMemberOf(cpGroupId);
+
+                if (leader != null)
+                {
+                    return Cluster.Messaging.SendToMemberAsync(request, leader.Uuid);
+                }
+            }
+            else
+            {
+                if (Cluster.State.Options.Networking.CPDirectToLeaderEnabled)
+                    Logger.IfWarning()?.LogWarning("CP Direct To Leader is an Enterprise feature, the request will be sent to any member");
```

Comment:
I'm afraid there is no proper way at the moment because there is config check during starting and even I check it on authentication that exceptions doesn't bubble up to user level. We need a moderate refactoring to have that.